lib/node/rpc.js

```diff
@@ -2533,6 +2538,67 @@ class RPC extends RPCBase {
     return null;
   }
 
+ async dumpzone(args, help) {
+    if (help || args.length !== 1)
+      throw new RPCError(errs.MISC_ERROR, 'dumpzone <filename>');
+
+    const valid = new Validator(args);
+    const filename = valid.str(0, null);
+    if (filename == null)
+      throw new RPCError(errs.MISC_ERROR, 'dumpzone <filename>');
+
+    const tmp = filename + '~';
+    this.logger.debug(`dumping root zone to file: ${filename}`);
+
+    let fd = null;
+    let fileErr = null;
+    fd = fs.createWriteStream(tmp, { flags: 'a+' });
+    fd.on('error', (err) => {
+      fd.end();
+      fd = null;
+      fileErr = err;
+    });
+
+    const tree = this.chain.db.tree;
+    const iter = tree.iterator(true);
+
+    let count = 0;
+    while ((await iter.next()) && (fd != null)) {
```

Comment:
Is this `fd` check redundant with the check on 2571?

Reply:
probably can remove one or other, just trying to catch if the `fd.on('error',...` got triggered 

JS isn't my strongest skill, so likely all that error handling can all be done better
I think it was to try & catch things like disk-full